Mage.Sets/src/mage/cards/a/AnheloThePainter.java

```diff
@@ -0,0 +1,166 @@
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.VAMPIRE, SubType.ASSASSIN);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Deathtouch
+        this.addAbility(DeathtouchAbility.getInstance());
+
+        // The first instant or sorcery spell you cast each turn has casualty 2.
+        this.addAbility(
+                new SimpleStaticAbility(
+                        Zone.BATTLEFIELD,
+                        new AnheloThePainterGainCausalityEffect()),
+                new AnheloThePainterWatcher()
+        );
+    }
+
+    private AnheloThePainter(final AnheloThePainter card) {
+        super(card);
+    }
+
+    @Override
+    public AnheloThePainter copy() {
+        return new AnheloThePainter(this);
+    }
+}
+
+class AnheloThePainterGainCausalityEffect extends ContinuousEffectImpl {
+
+
+    AnheloThePainterGainCausalityEffect() {
+        super(Duration.WhileOnBattlefield, Layer.AbilityAddingRemovingEffects_6, SubLayer.NA, Outcome.AddAbility);
+        staticText = "The first instant or sorcery spell you cast each turn has casualty 2. " +
+                "<i>(As you cast that spell, you may sacrifice a creature with power 2 or greater. " +
+                "When you do, copy the spell and you may choose new targets for the copy.)</i>";
+    }
+
+    AnheloThePainterGainCausalityEffect(final AnheloThePainterGainCausalityEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        AnheloThePainterWatcher watcher = game.getState().getWatcher(AnheloThePainterWatcher.class);
+        if (controller == null || watcher == null) {
+            return false;
+        }
+
+        // Get all instant and socery cards owned by the player
+        Stream<Card> cardStream = Stream.concat(
+                controller.getLibrary().getCards(game).stream().filter(Card::isInstantOrSorcery),
+                controller.getGraveyard().getCards(game).stream().filter(Card::isInstantOrSorcery));
+        cardStream = Stream.concat(cardStream,
+                game.getExile().getAllCards(game, controller.getId()).stream().filter(Card::isInstantOrSorcery));
+        cardStream =  Stream.concat(cardStream,
+                controller.getHand().getCards(game).stream().filter(Card::isInstantOrSorcery));
+        // TODO: This still isn't good since it must still capture opponents cards that the controller may get access to.
+
+        Iterator<Card> cardIterator = cardStream.iterator();
+        while (cardIterator.hasNext()) {
+            Card card = cardIterator.next();
+            // TODO: I know this doesn't work, not sure how to check this
+            if (FirstInstantOrSorceryEachTurnCondition.instance.apply(game, source)) {
+                game.getState().addOtherAbility(card, new CasualtyAbility(card, 2));
+            }
```

Comment:
This is a mess. I am not sure how to fix it from here. Causality is an optional additional cost, so I have to add it to the card *before* the card's been cast.

Reply:
Also need to take card of MDFCs?